### ISSUE TITLE
Support reading from Java bean sources

### ISSUE THREAD

**Issue by krzemin** [scalalandio/chimney#71]
Having Java bean-like class:

```
class Foo(private var id: Int, private var name: String) {
  def getId: Int = id
  def getName: String = name
}
```

we would like to be able to transform it to a case class:

```
case class Bar(id: Int, name: String)
```

We cannot simply access fields, because they're private, but we can add support for interpreting getters with `getXyz` name as sources for `xyz` fields.

**Comment by jhalterman** [comment]
One ask of this feature is that we be able to support properties that adhere to different naming conventions. The JavaBeans convention expects accessors to be prefixed with `get` or `is`, ex: `String getFoo()` or `boolean isFoo()`. But often Java data classes have accessors that lack those prefixes, ex: `String foo()`.

Additionally, it would be nice if `Optional` return values were supported, ex: `Optional<String> getFoo()`.

**Comment by krzemin** [comment]
@jhalterman absolutely! Accessors lacking those prefixes should be supported even now. Support for Java `Optional<T>` (convertibility to scala `Option[T]`) would be really nice to have too. The same could be implemented for Java collections.

Wondering if possibility to disable awareness of Java beans convention would be useful (like we have with [default values](https://scalalandio.github.io/chimney/#Disablingdefaultvalues))?

**Comment by jhalterman** [comment]
> Wondering if possibility to disable awareness of Java beans convention would be useful 

I don't have an opinion on that. What would be the benefit of disabling?

**Comment by kuba-mikulski** [comment]
Added support for automatic getter lookup (get/is) in https://github.com/scalalandio/chimney/pull/81, can be disabled with `disableBeanGetterLookup`. Note that the lookup is based solely on name and signature, so theoretically if the source has a getter-like method that actually performs some side effects, we have no way of verifying that - that's why the option to disable would be useful.

**Comment by kuba-mikulski** [comment]
Also, Optional -> Option not done yet, I can create a separate issue for that.

**Comment by krzemin** [comment]
Merged in #81. For optionals and few other java interop transformations I created separate issue: #86
